src/interface.jl

```diff
@@ -232,15 +232,22 @@ function coloring(
 ) where {decompression,R}
     m, n = size(A)
     T = eltype(A)
-    Aᵀ = if symmetric_pattern || A isa Union{Symmetric,Hermitian}
-        A
-    else
-        transpose(A)
-    end  # TODO: fuse with next step?
-    A_and_Aᵀ = [
-        spzeros(T, n, n) SparseMatrixCSC(Aᵀ)
-        SparseMatrixCSC(A) spzeros(T, m, m)
-    ]  # TODO: slow
+
+    # build A_and_Aᵀ
+    sparse_A = A isa SparseMatrixCSC ? A : SparseMatrixCSC(A)
+    nnzA = nnz(A)
```

Comment:
```suggestion
    nnzA = nnz(sparse_A)
```
I think that's why the tests fail